File: PharmacySrenity.Web/Modules/ServerTypes/Province/ProvinceRow.ts
import { fieldsProxy } from "@serenity-is/corelib";

export interface ProvinceRow {
    Id?: number;
    Name?: string;
}

export abstract class ProvinceRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'Province.Province';
    static readonly deletePermission = 'Province:Modify';
    static readonly insertPermission = 'Province:Modify';
    static readonly readPermission = 'Province:View';
    static readonly updatePermission = 'Province:Modify';

    static readonly Fields = fieldsProxy<ProvinceRow>();
}

File: PharmacySrenity.Web/Modules/Province/Province/ProvinceRow.cs
using DocumentFormat.OpenXml.Wordprocessing;
using PharmacySrenity.Modules.Province.Province;

namespace PharmacySrenity.Province;

[ConnectionKey("Default"), Module("Province"), TableName("Province")]
[DisplayName("Province"), InstanceName("Province")]
[ReadPermission(ProvincePermissionKeys.View)]
[ModifyPermission(ProvincePermissionKeys.Modify)]
[ServiceLookupPermission("General")]
public sealed class ProvinceRow : Row<ProvinceRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Name"), Size(100), NotNull, QuickSearch, NameProperty]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField Name;

    }
}